lib/grack/file_streamer.rb

```diff
@@ -0,0 +1,41 @@
+require 'pathname'
+
+require 'grack/io_streamer'
+
+module Grack
+  ##
+  # A Rack body implementation that streams a given file in chunks for a Rack
+  # response.
+  class FileStreamer < IOStreamer
+    ##
+    # Creates a new instance of this object.
+    #
+    # @param [Pathname, String] path a path to a file.
+    def initialize(path)
+      @path = Pathname.new(path).expand_path
+    end
+
+    ##
+    # In order to support X-Sendfile when available, this method returns the
```

Comment:
From what I recall when I wrote that stuff, Rack looks at the response body, and if the body responds to `#to_path`, Rack adds the appropriate headers.  I haven't had a chance to try that out myself though to make sure it really works.